src/D2L.CodeStyle.Analyzers/Immutability/TypeDeclarationImmutabilityAnalyzer.cs

```diff
@@ -128,13 +128,29 @@ INamedTypeSymbol typeSymbol
 			GenericNameSyntax syntax
 		) {
 			SymbolInfo info = ctx.SemanticModel.GetSymbolInfo( syntax, ctx.CancellationToken );
-			if( !( info.Symbol is IMethodSymbol method ) ) {
-				return;
+
+			ImmutableArray<ITypeParameterSymbol> typeParameters;
+			ImmutableArray<ITypeSymbol> typeArguments;
+			switch( info.Symbol ) {
+				case IMethodSymbol method:
+					typeParameters = method.TypeParameters;
+					typeArguments = method.TypeArguments;
+					break;
+				case INamedTypeSymbol namedType:
+					typeParameters = namedType.TypeParameters;
+					typeArguments = namedType.TypeArguments;
+					break;
+				default:
+					return;

```

Comment:
nit: helper method? 

```cs
private static (
  ImmutableArray<ITypeParameterSymbol> TypeParameters,
  ImmutableArray<ITypeSymbol> TypeArguments
) GetTypeParamsAndArgs( ITypeSymbol type )
  => type switch {
    IMethodSymbol method => (method.TypeParameters, method.TypeArguments),
    INamedTypeSymbol namedType => (namedType.TypeParameters, namedType.TypeArguments)
    _ => throw new NotImplementedException();
  }
```

or something

Reply:
Done. Not positive if there can be other types of symbols. But our spec test didn't throw at the very least